rust/src/lib.rs

```diff
@@ -40,22 +39,35 @@ use treefile::treefile_read_impl;
 
 /* Wrapper functions for translating from C to Rust */
 
+/// Convert a C (UTF-8) string to a &str; will panic
+/// if it isn't valid UTF-8.  Note the lifetime of
+/// the return value must be <= the pointer.
+fn str_from_nullable<'a>(s: *const libc::c_char) -> Option<&'a str> {
+    if s.is_null() {
+        None
+    } else {
+        let s = unsafe { CStr::from_ptr(s) };
+        Some(s.to_str().unwrap())
+    }
+}
+
 #[no_mangle]
 pub extern "C" fn rpmostree_rs_treefile_read(
     filename: *const libc::c_char,
+    arch: *const libc::c_char,
     fd: libc::c_int,
     error: *mut *mut glib_sys::GError,
 ) -> libc::c_int {
+    // Convert arguments
+    let filename = Path::new(unsafe { CStr::from_ptr(filename).to_str().unwrap() });
```

Comment:
Not that we really want to cater to the anti-UTF-8 rebels, though what was wrong with the previous way of building a `Path` which supported any path?

Reply:
Yeah my feeling is that the `OsStr` stuff is mostly for tools like `rm` or `du` that don't actually need to interpret filenames in general.  I'm totally fine erroring out if the user provides us a treefile whose filename isn't valid UTF-8.  That said we shouldn't panic.  I reworked this some :arrow_down: 

Also holy cow is the `*const libc::c_char/CStr/[u8]/OsStr/Path` situation confusing...I was able to drop the `Path::new` as we don't need/want a mutable buffer.